### PR TITLE
feat: defer `commitBody` compile

### LINKED ISSUE #17063: Defer commitBody compilation
### Describe the proposed change(s).

Currently we compile `commitBody` into `commitMessage` fairly early, during branch update generation: https://github.com/renovatebot/renovate/blob/4ead106f81accce473c0bc73b05c1c4dc738a127/lib/workers/repository/updates/generate.ts#L221-L226

Instead, we should defer `commitBody` compilation until we know we need to do a commit.

Note: The first line of `commitMessage` still needs compilation early because it's used for `prTitle`.

### PR DESCRIPTION
**Pull request by viceice**
## Changes
- defer

## Context
- closes #17063
- https://github.com/renovate-reproductions/docker-node-versioning-issue/pull/1/commits/f09bdb3d8044e03216c50c8f8a6228e3ebe098e6
- https://github.com/renovate-reproductions/docker-node-versioning-issue/pull/2/commits/7217346e9daa176b385db8022605ffed35c79367

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/workers/repository/update/branch/index.spec.ts [hunk 1]
```diff
@@ -656,13 +657,13 @@ describe('workers/repository/update/branch/index', () => {
...
           prCreation: 'not-pending',
+          commitBody: '[skip-ci]',
         })
       ).toEqual({
         branchExists: true,
         prNo: undefined,
         result: 'pending',
       });
-
       expect(automerge.tryBranchAutomerge).toHaveBeenCalledTimes(0);
```
Comment (viceice) [inline lib/workers/repository/update/branch/index.spec.ts:667]:
```suggestion

      expect(automerge.tryBranchAutomerge).toHaveBeenCalledTimes(0);
```

File: lib/workers/repository/changelog/index.ts [hunk 2]
```diff
@@ -20,8 +22,12 @@ export async function embedChangelogs(
...
-  for (const field of exposedConfigOptions) {
+export function needsChangelogs(
+  upgrade: BranchUpgradeConfig,
+  fields = exposedConfigOptions.filter((o) => o !== 'commitBody')
+): boolean {
+  // commitBody is now compiled when commit is done
+  for (const field of fields) {
     // fields set by `getChangeLogJSON`
     if (containsTemplates(upgrade[field], ['logJSON', 'releases'])) {
       return true;
```
Comment (rarkins) [inline lib/workers/repository/changelog/index.ts:33]: Should this be a warn now?

Comment (viceice) [reply lib/workers/repository/changelog/index.ts:33]: it's here
https://github.com/renovatebot/renovate/blob/6fe3f2ff326ed4136a6a1d529407997a55815c03/lib/workers/repository/updates/branchify.ts#L79-L88

Comment (viceice) [reply lib/workers/repository/changelog/index.ts:33]: i need this fuction later here

https://github.com/renovatebot/renovate/blob/6fe3f2ff326ed4136a6a1d529407997a55815c03/lib/workers/repository/update/branch/index.ts#L482-L499

**Review by rarkins — APPROVED** [review]

**Pull request merged**